plugin/advisor/mysql/advisor_index_type_no_blob.go

```diff
@@ -0,0 +1,227 @@
+package mysql
+
+// Framework code is generated by the generator.
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pingcap/tidb/parser/ast"
+	"github.com/pingcap/tidb/parser/mysql"
+	"github.com/pingcap/tidb/parser/types"
+	"github.com/pkg/errors"
+
+	"github.com/bytebase/bytebase/plugin/advisor"
+	"github.com/bytebase/bytebase/plugin/advisor/catalog"
+	"github.com/bytebase/bytebase/plugin/advisor/db"
+)
+
+var (
+	_ advisor.Advisor = (*IndexTypeNoBlobAdvisor)(nil)
+	_ ast.Visitor     = (*indexTypeNoBlobChecker)(nil)
+)
+
+func init() {
+	advisor.Register(db.MySQL, advisor.MySQLIndexTypeNoBlob, &IndexTypeNoBlobAdvisor{})
+	advisor.Register(db.TiDB, advisor.MySQLIndexTypeNoBlob, &IndexTypeNoBlobAdvisor{})
+}
+
+// IndexTypeNoBlobAdvisor is the advisor checking for index type no blob.
+type IndexTypeNoBlobAdvisor struct {
+}
+
+// Check checks for index type no blob.
+func (*IndexTypeNoBlobAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	stmtList, errAdvice := parseStatement(statement, ctx.Charset, ctx.Collation)
+	if errAdvice != nil {
+		return errAdvice, nil
+	}
+
+	level, err := advisor.NewStatusBySQLReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+	checker := &indexTypeNoBlobChecker{
+		level:            level,
+		title:            string(ctx.Rule.Type),
+		database:         ctx.Database,
+		tablesNewColumns: make(map[string]columnNameToColumnDef),
+	}
+
+	for _, stmt := range stmtList {
+		checker.text = stmt.Text()
+		checker.line = stmt.OriginTextPosition()
+		(stmt).Accept(checker)
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    advisor.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+	return checker.adviceList, nil
+}
+
+type indexTypeNoBlobChecker struct {
+	adviceList       []advisor.Advice
+	level            advisor.Status
+	title            string
+	text             string
+	line             int
+	database         *catalog.Database
+	tablesNewColumns tableNewColumn
+}
+
+// Enter implements the ast.Visitor interface.
+func (v *indexTypeNoBlobChecker) Enter(in ast.Node) (ast.Node, bool) {
+	var pkDataList []pkData
+	switch node := in.(type) {
+	case *ast.CreateTableStmt:
```

Comment:
We need to consider the CREATE INDEX statement.

Reply:
Done. PTAL.